pkg/solana/chainreader/chain_reader.go

```diff
@@ -217,54 +227,99 @@ func (s *SolanaChainReaderService) CreateContractType(readIdentifier string, for
+func (s *SolanaChainReaderService) init(namespaces map[string]config.ChainContractReader) error {
+	for namespace, nameSpaceDef := range namespaces {
+		for genericName, read := range nameSpaceDef.Reads {
+			injectAddressModifier(read.InputModifications, read.OutputModifications)
+			idlDef, err := codec.FindDefinitionFromIDL(codec.ChainConfigTypeAccountDef, read.ChainSpecificName, nameSpaceDef.IDL)
 			if err != nil {
 				return err
 			}
 
-			codecWithModifiers, err := codec.NewNamedModifierCodec(idlCodec, procedure.IDLAccount, mod)
-			if err != nil {
-				return err
+			switch read.ReadType {
+			case config.Account:
+				accountIDLDef, isOk := idlDef.(codec.IdlTypeDef)
+				if !isOk {
+					return fmt.Errorf("unexpected type %T from IDL definition for account read: %q, with chainSpecificName: %q, of type: %q", accountIDLDef, genericName, read.ChainSpecificName, read.ReadType)
+				}
+				if err = s.addAccountRead(namespace, genericName, nameSpaceDef.IDL, accountIDLDef, read); err != nil {
+					return err
+				}
+			case config.Event:
+				eventIDlDef, isOk := idlDef.(codec.IdlEvent)
+				if !isOk {
+					return fmt.Errorf("unexpected type %T from IDL definition for log read: %q, with chainSpecificName: %q, of type: %q", eventIDlDef, genericName, read.ChainSpecificName, read.ReadType)
+				}
+				// TODO s.addLogRead()
+				return fmt.Errorf("implement me")
+			default:
+				return fmt.Errorf("unexpected read type %q for: %q in namespace: %q", read.ReadType, genericName, namespace)
 			}
-
-			s.bindings.AddReadBinding(namespace, methodName, newAccountReadBinding(
-				procedure.IDLAccount,
-				codecWithModifiers,
-				createRPCOpts(procedure.RPCOpts),
-			))
 		}
 	}
 
 	return nil
 }
 
+func (s *SolanaChainReaderService) addAccountRead(namespace string, genericName string, idl codec.IDL, idlType codec.IdlTypeDef, readDefinition config.ReadDefinition) error {
+	inputAccountIDLDef := codec.NilIdlTypeDefTy
+	// TODO:
+	//		if hasPDA{
+	//			inputAccountIDLDef = pdaType
+	//		}
+	if err := s.addCodecDef(true, namespace, genericName, codec.ChainConfigTypeAccountDef, idl, inputAccountIDLDef, readDefinition.InputModifications); err != nil {
+		return err
+	}
+
+	if err := s.addCodecDef(false, namespace, genericName, codec.ChainConfigTypeAccountDef, idl, idlType, readDefinition.OutputModifications); err != nil {
+		return err
+	}
+
+	s.lookup.addReadNameForContract(namespace, genericName)
+
+	s.bindings.AddReadBinding(namespace, genericName, newAccountReadBinding(
+		namespace,
+		genericName,
+		createRPCOpts(readDefinition.RPCOpts),
+	))
+
+	return nil
+}
+
 // injectAddressModifier injects AddressModifier into OutputModifications.
 // This is necessary because AddressModifier cannot be serialized and must be applied at runtime.
-func injectAddressModifier(outputModifications codeccommon.ModifiersConfig) {
+func injectAddressModifier(inputModifications, outputModifications commoncodec.ModifiersConfig) {
```

Comment:
Can you elaborate on this a little bit more? Why would someone use `commoncodec.AddressBytesToStringModifierConfig` instead of `codec.SolanaAddressModifier{}`?

Reply:
Because this  `AddressBytesToStringModifierConfig` is the actual serializeable  modifier config, the actual chain specific impl. which is a function and therefore cannot be serialized is injected during CR startup